Change transforms to support grayscale images

deep-jkl opened the pull request:
Image.fromarray cannot load image of shape (h, w, 1) without format specification. Forcing to use "L" solves it.


This could also fix #1020 considering that backbone can process single channel images.

deep-jkl commented:
I suggest to change tests and expand dimension of input test images to 3 (including channel).

ppwwyyxx commented:
Thanks for the fix!

Since both `(h,w)` and `(h,w,1)` should be supported, could you add a new test that checks both shapes can be used in Resize & Extent? This would be better than changing existing `(h,w)` to `(h,w,1)`.

deep-jkl commented:
> Thanks for the fix!
> 
> Since both `(h,w)` and `(h,w,1)` should be supported, could you add a new test that checks both shapes can be used in Resize & Extent? This would be better than changing existing `(h,w)` to `(h,w,1)`.

Done, I changed the existing tests back and added tests for Resize and Extent. Tests are testing three possible shapes (h, w), (h, w, 1), (h, w, 3).
